packages/clay-date-picker/src/index.tsx

```diff
@@ -344,35 +344,68 @@ const ClayDatePicker: React.FunctionComponent<IProps> = React.forwardRef<
 		const inputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
 			const {value} = event.target;
 
-			const format = time
-				? `${dateFormat} ${use12Hours ? TIME_FORMAT_12H : TIME_FORMAT}`
-				: dateFormat;
+			if (!value) {
```

Comment:
For that case when you don't have a value in the input, you can simplify the implementation here to just reset to the initial values. For example, we don't need to call `onValueChange` and do `dateFormatted` because in that case, we expect the state to be empty.

- Instead of using `NEW_DATE` directly, you can use `initialMonth` because it can be configured by the developer.
- I imagine you only need to call `changeMonth` with `initialMonth`, `setDaysSelected ` and `setCurrentTime` if the time is enabled.